lib/airtable/entity/record.rb

```diff
@@ -0,0 +1,94 @@
+require 'time'
+# Main object for store Airtable Record entity
+module Airtable
+  module Entity
+    # Airtable Record entity
+    class Record
+      extend Forwardable
+      attr_reader :id, :created_at, :fields
+
+      def_delegators :@fields, :[], :[]=
+
+      def initialize(id, options = {})
+        @id = id
+        parse_options(options)
+      end
+
+      def new_record?
+        @id.nil? || @id.empty?
+      end
+
+      def __create__(base, name)
+        res = base.__make_request__(:post, name, fields: fields)
+        @id = res['id']
+        parse_options(fields: res['fields'], created_at: res['createdTime'])
+        self
+      end
+
+      def __update__(base, name)
+        args = [:patch, [name, @id].join('/'), fields: fields]
+        res  = base.__make_request__(*args)
+        parse_options(fields: res['fields'])
+        self
+      end
+
+      def __fetch__(base, path)
+        res = base.__make_request__(:get, path, {})
+        parse_options(fields: res['fields'], created_at: res['createdTime'])
+        self
+      end
+
+      def __replace__(base, name)
+        res = base.__make_request__(:put, [name, @id].join('/'), fields: fields)
+        parse_options(fields: res['fields'])
+        self
+      end
+
+      def __destroy__(base, name)
+        res = base.__make_request__(:delete, [name, @id].join('/'), {})
+        res['deleted']
+      end
+
+      def [](key)
+        @fields[key.to_s]
+      end
+
+      def []=(key, value)
+        @fields[key.to_s] = value
+      end
+
+      class << self
+        def all(base, name, params)
+          res = []
+          __fetch__(base, name, params, res)
+          res
+        end
+
+        private
+
+        def __fetch__(base, name, params, res)
+          result = base.__make_request__(:get, name, params)
+          result['records'].each do |r|
+            args = [
+              r['id'], fields: r['fields'], created_at: r['createdTime']
+            ]
+            res << new(*args)
+          end
+          return unless result['offset']
+          __fetch__(base, name, params.merge(offset: result['offset']), res)
```

Comment:
lets add a comment that ruby can easily handle 10k deep stack and that should be plenty to fetch any table.

Reply:
You mean to add to README?

Reply:
no, just here. recursion is a bit of a red flag unless the language always does tail call optimization (ruby has an option that's off by default). Hence, it helps to add a comment to the implementation justifying why it shouldn't cause trouble. There are esoteric scenarios where this will blow up. E.g. iterating over a large base (50k records) with page_size set to 1 (really bad idea of course).